### DISCUSSION TITLE
useAsyncData returning null on page reload when deployed

### DISCUSSION THREAD

**Discussion by andyjamesn** [nuxt/nuxt#17172] [Questions]
This works fine while developing locally but not after I deploy to production.

```html
<script setup>
const styleurl = `https://dev--supabase-style-guide-api.globalpress.autocode.gg/style/`;
const stylefetchOptions = { method: "POST", body: { slug: 'america-american', lang: 'en' } };
const { data: style } = await useAsyncData('style', () => $fetch(styleurl, stylefetchOptions));

// You can see the value inside the style object in dev and prod
console.log(style);

// In dev it logs out style.value but after deployed to production style.value value is null
console.log(style.value);
</script>
```

It works fine if I link to the page using NuxtLink but not on page reload.

I can render {{style}} in the template too, but not when I try access `style.value` inside setup. I need to get an ID inside style to do another useFetch call inside setup.

You can see the result in the console logs of this app

https://delightful-longma-17ca34.netlify.app/en/style/america-american

Here is what it is logging out and the difference between local and deployed.

**Local development**
<img width="1028" alt="1dev" src="https://user-images.githubusercontent.com/54320410/181819594-87231e10-d66f-4035-96f2-57e203f11714.png">

**Live deployed**
<img width="979" alt="2server" src="https://user-images.githubusercontent.com/54320410/181819608-e960f38e-96df-477d-b7dd-750e147224ed.png">

**Comment by danielroe** [comment]
Would you try with the latest edge release?

**Reply by andyjamesn** [reply]
Sure, I have not used the latest edge release before but I think I know how..

Can you confirm, do I change

```
"devDependencies": {
    "nuxt": "3.0.0-rc.6"
 }
```

To

```
"devDependencies": {
    "nuxt": "npm:nuxt3@latest"
  }
```

Then run yarn install?

**Reply by andyjamesn** [reply]
@danielroe I did the above and deployed and the build log shows

```
8:29:38 PM: $ nuxt build
8:46:38 PM: [log] Nuxt CLI v3.0.0-rc.6-27651720.5dc864d
```

So it looks like it is using the latest edge release. 

But unfortunately the issue persists.

**Comment by andyjamesn** [comment]
After spending some more time with this I wanted to post some findings. 

I am not sure if it is a bug or if its intended to work this way and I am just doing it wrong.

First, if I set `ssr: false` in my nuxt config and then build, the result logs just fine. But since I want it to be server rendering this did not really help.

So after reading this article I decided to try `refreshNuxtData()`

https://v3.nuxtjs.org/guide/features/data-fetching/

I simply added `refreshNuxtData` after the first useAsyncData as follows

```js
const styleurl = `https://dev--supabase-style-guide-api.globalpress.autocode.gg/style/`;
const stylefetchOptions = { server: false, method: "POST", body: { slug: 'abortion', lang: 'en' } };
const { data: style } = await useAsyncData('style', () => $fetch(styleurl, stylefetchOptions));

//  Refresh Data //
await refreshNuxtData('style'); // <--- this what I added

console.log(style.value); // Logging style.value now works as expected
```

Now I can do another useAsyncData using the style from the previous useAsyncData

```JS
const fetchOptions = { body: { styleId: style.value.object.objectID } }; // styleId comes from previous asyncdata 
const { data: questions } = await useAsyncData(
    'questions',
    () => $fetch(url, fetchOptions)
)
```

Any insight into this? Was I just doing it wrong?
